src/main/java/org/ibp/api/java/impl/middleware/file/validator/FileValidator.java

```diff
@@ -2,36 +2,84 @@
 
 import org.apache.commons.io.FilenameUtils;
 import org.apache.commons.lang3.StringUtils;
-import org.ibp.api.Util;
+import org.generationcp.middleware.api.brapi.v1.image.ImageNewRequest;
+import org.generationcp.middleware.api.brapi.v2.germplasm.ExternalReferenceDTO;
 import org.ibp.api.exception.ApiRequestValidationException;
+import org.ibp.api.java.file.FileStorageService;
+import org.ibp.api.java.impl.middleware.common.validator.BaseValidator;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
 import org.springframework.validation.BindingResult;
+import org.springframework.validation.MapBindingResult;
 import org.springframework.web.multipart.MultipartFile;
 
+import java.util.HashMap;
 import java.util.List;
 
+import static org.ibp.api.java.impl.middleware.common.validator.BaseValidator.checkNotNull;
+
 @Component
 public class FileValidator {
+	private static final int MAX_REFERENCE_ID_LENGTH = 2000;
+	private static final int MAX_REFERENCE_SOURCE_LENGTH = 255;
+
+	@Autowired
+	private FileStorageService fileStorageService;
 
 	@Value("#{'${file.upload.supported.types}'.toLowerCase().split(',')}")
 	private List<String> supportedFileTypes;
 
-	public void validateFile(final BindingResult errors, final MultipartFile file) {
+	protected BindingResult errors;
+
+	public void validateFile(final MultipartFile file) {
+		this.errors = new MapBindingResult(new HashMap<>(), String.class.getName());
 		if (file == null) {
-			errors.reject("file.upload.required", "");
-			throw new ApiRequestValidationException(errors.getAllErrors());
+			this.errors.reject("file.upload.required", "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
 		}
 
-		this.validateExtension(errors, file.getOriginalFilename());
+		this.validateExtension(file.getOriginalFilename());
 	}
 
-	public void validateExtension(final BindingResult errors, final String fileName) {
+	public void validateExtension(final String fileName) {
 		final String extension = FilenameUtils.getExtension(fileName);
 		if (!this.supportedFileTypes.contains(extension.toLowerCase())) {
-			errors.reject("file.upload.not-supported",  new String[] {StringUtils.join(this.supportedFileTypes, ", ")}, "");
-			throw new ApiRequestValidationException(errors.getAllErrors());
+			this.errors.reject("file.upload.not-supported",  new String[] {StringUtils.join(this.supportedFileTypes, ", ")}, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
 		}
 	}
 
+	public void validateFileStorage() {
+		BaseValidator.checkArgument(this.fileStorageService.isConfigured(), "file.storage.not.configured");
+	}
+
+	public void validateImage(final ImageNewRequest body) {
```

Comment:
@mderamos this function is very specific for BRAPI. It will me much clear to keep FileValidator BrAPI agnostic and move this validateImage to a new validator class or to the resource